src/zenml/zen_stores/migrations/versions/ec6307720f92_simplify_model_version_links.py

```diff
@@ -0,0 +1,112 @@
+"""Simplify model version links [ec6307720f92].
+
+Revision ID: ec6307720f92
+Revises: 1cb6477f72d6
+Create Date: 2024-11-06 16:16:43.344569
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "ec6307720f92"
+down_revision = "1cb6477f72d6"
+branch_labels = None
+depends_on = None
+
+
+def _migrate_artifact_type() -> None:
+    """Migrate the artifact type."""
+    meta = sa.MetaData()
+    meta.reflect(
+        bind=op.get_bind(),
+        only=("artifact_version", "model_versions_artifacts"),
+    )
+    artifact_version_table = sa.Table("artifact_version", meta)
+    model_versions_artifacts = sa.Table("model_versions_artifacts", meta)
+
+    connection = op.get_bind()
+
+    query = sa.select(
+        model_versions_artifacts.c.artifact_version_id,
+        model_versions_artifacts.c.is_model_artifact,
+        model_versions_artifacts.c.is_deployment_artifact,
+    )
+    result = connection.execute(query)
+
+    updated = set()
+    for (
+        artifact_version_id,
+        is_model_artifact,
+        is_deployment_artifact,
+    ) in result:
+        if artifact_version_id in updated:
+            # If an artifact was a model artifact in one model version, and a
+            # deployment artifact in another model version, we only update it
+            # once.
```

Comment:
So it will be something random, right?

Reply:
Yes, this will be random, depends on which input artifact connection is read first. This should be a very uncommon case though, I think it should be fine?